satpy/modifiers/parallax.py

```diff
@@ -0,0 +1,495 @@
+        """Calculate the corrected lon/lat based from the shifted area.
+
+        After calculating the shifted area based on :func:`forward_parallax`,
+        we invert the parallax error and estimate where those pixels came from.
+        For details on the algorithm, see the class docstring.
+        """
+        (shifted_lon, shifted_lat) = shifted_area.get_lonlats(chunks=1024)
+        lon_diff = shifted_lon - base_lon
+        lat_diff = shifted_lat - base_lat
+        # We use the bucket resampler here, because parallax correction
+        # inevitably means there will be 2 source pixels ending up in the same
+        # destination pixel.  We want to choose the biggest shift (max abs in
+        # lat_diff and lon_diff), because the biggest shift corresponds to the
+        # highest clouds, and if we move a 10 km cloud over a 2 km one, we
+        # should retain the 10 km.
+        #
+        # some things to keep in mind:
+        # - even with a constant cloud height, 3 source pixels may end up in
+        #   the same destination pixel, because pixels get larger in the
+        #   direction of the satellite.  This means clouds may shrink as they
+        #   approach the satellite.
+        # - the x-shift is a function of y and the y-shift is a function of x,
+        #   so a cloud that was rectangular at the start may no longer be
+        #   rectangular at the end
+        br = BucketResampler(self.base_area,
+                             da.array(shifted_lon), da.array(shifted_lat))
+        inv_lat_diff = br.get_abs_max(lat_diff)
+        inv_lon_diff = br.get_abs_max(lon_diff)
+
+        inv_lon = base_lon - inv_lon_diff
+        inv_lat = base_lat - inv_lat_diff
+        if self.debug_mode:
+            self.diagnostics["shifted_lon"] = shifted_lon
+            self.diagnostics["shifted_lat"] = shifted_lat
+            self.diagnostics["inv_lon"] = inv_lon
+            self.diagnostics["inv_lat"] = inv_lat
+            self.diagnostics["inv_lon_diff"] = inv_lon_diff
+            self.diagnostics["inv_lat_diff"] = inv_lat_diff
+            self.diagnostics["base_lon"] = base_lon
+            self.diagnostics["base_lat"] = base_lat
+            self.diagnostics["lon_diff"] = lon_diff
+            self.diagnostics["lat_diff"] = lat_diff
+            self.diagnostics["shifted_area"] = shifted_area
+            self.diagnostics["count"] = xr.DataArray(
+                br.get_count(), dims=("y", "x"), attrs={"area": self.base_area})
+        return (inv_lon, inv_lat)
+
+
+class ParallaxCorrectionModifier(ModifierBase):
+    """Modifier for parallax correction.
+
+    Apply parallax correction as a modifier.  Uses the
+    :class:`ParallaxCorrection` class, which in turn uses the
+    :func:`forward_parallax` function.  See the documentation there for
+    details on the behaviour.
+
+    To use this, add in your ``etc/modifiers/visir.yaml`` something like::
```

Comment:
Shouldn't this rather be `etc/composites/visir.yaml`

Reply:
Yes, definitely.

Reply:
Thanks, fixed.